gregorio-test.sh

```diff
@@ -52,6 +53,14 @@ then
     source $rcfile
 fi
 
+if [[ "$OSTYPE" == "linux-gnu" ]]
+then
+	export NPROCESSORS="${NPROCESSORS:-$(nproc)}"
+elif [[ "$OSTYPE" == "darwin"* ]]
+then
+	export NPROCESSORS="${NPROCESSORS:-$(sysctl -n hw.ncpu)}"
+fi
```

Comment:
I should have realized it before, but there should be an "else" here that simply defaults NPROCESSORS to 1 if not previously set.  If the system is neither linux-gnu nor darwin, the later line will not work.

Reply:
I'll change it, but wouldn't that effectively be the situation we have now where `$processors` is not defined?

Reply:
Line 442:
```
processors=$(nproc 2>/dev/null || echo 1)
```
This sets processors to 1 if the nproc command fails.

Since you changed `processors` to `NPROCESSORS` below, then `NPROCESSORS` needs to behave the same because the `-P` option to xargs requires an argument.

This also makes it obvious to me that the line that sets `processors` is no longer needed and should be deleted.

Reply:
How did I miss that line earlier?  Changes coming up.